### ISSUE TITLE
Configure AWS for local development: An error occurred (AccessDenied) when calling the ListBuckets operation: User: arn:aws:sts::<...>:assumed-role/AWSReservedSSO_amplify-policy_<...>/amplify-admin is not authorized to perform: s3:ListAllMyBuckets because no identity-based policy allows the s3:ListAllMyBuckets action

### ISSUE THREAD

**Issue by tomphan** [aws-amplify/amplify-backend#2308]
### Environment information

```plain text
System:
  OS: Windows 10 10.0.19045
  CPU: (8) x64 Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
  Memory: 3.98 GB / 15.89 GB
Binaries:
  Node: 22.9.0 - C:\Program Files\nodejs\node.EXE
  Yarn: 1.22.22 - ~\AppData\Roaming\npm\yarn.CMD
  npm: 10.9.0 - C:\Program Files\nodejs\npm.CMD
  pnpm: undefined - undefined
NPM Packages:
  @aws-amplify/auth-construct: 1.3.1
  @aws-amplify/backend: 1.5.0
  @aws-amplify/backend-auth: 1.2.0
  @aws-amplify/backend-cli: 1.2.9
  @aws-amplify/backend-data: 1.1.4
  @aws-amplify/backend-deployer: 1.1.5
  @aws-amplify/backend-function: 1.7.0
  @aws-amplify/backend-output-schemas: 1.3.0
  @aws-amplify/backend-output-storage: 1.1.2
  @aws-amplify/backend-secret: 1.1.4
  @aws-amplify/backend-storage: 1.2.1
  @aws-amplify/cli-core: 1.1.3
  @aws-amplify/client-config: 1.4.0
  @aws-amplify/deployed-backend-client: 1.4.2
  @aws-amplify/form-generator: 1.0.3
  @aws-amplify/model-generator: 1.0.8
  @aws-amplify/platform-core: 1.1.0
  @aws-amplify/plugin-types: 1.3.0
  @aws-amplify/sandbox: 1.2.3
  @aws-amplify/schema-generator: 1.2.4
  aws-amplify: 6.6.6
  aws-cdk: 2.163.1
  aws-cdk-lib: 2.163.1
  typescript: 5.6.3
AWS environment variables:
  AWS_NODEJS_CONNECTION_REUSE_ENABLED = 1
  AWS_STS_REGIONAL_ENDPOINTS = regional
No CDK environment variables
```

### Describe the bug

When following official tutorial "[Configure AWS for local development](https://docs.amplify.aws/react/start/account-setup/)", in the step: 
[5. Bootstrap your AWS account](https://docs.amplify.aws/react/start/account-setup/#5-bootstrap-your-aws-account)

Calling in the Terminal:
`aws s3 ls --profile default`

The error occures:

> An error occurred (AccessDenied) when calling the ListBuckets operation: User: arn:aws:sts::<...>:assumed-role/AWSReservedSSO_amplify-policy_<...>/amplify-admin is not authorized to perform: s3:ListAllMyBuckets because no identity-based policy allows the s3:ListAllMyBuckets action

Also calling:
`npx ampx sandbox --profile default`

Error displays:

> InvalidCredentialError: Failed to load default AWS credentials

Please can anyone have suggestion. Thanks.

### Reproduction steps

Just following a fresh, carefully tutorial steps: https://docs.amplify.aws/react/start/account-setup/

**Comment by tomphan** [comment]
Calling as guilded in the tutorial:
`aws s3 ls --profile default`

I think I do not have `S3 `setup in my AWS, in my case I'm trying `amplify`, not `S3`.

Please advise. Thx.

**Comment by ykethan** [comment]
Hey @tomphan, thank you for reaching. Could you confirm if the `C:\Users\USERNAME\.aws\config` and `C:\Users\USERNAME\.aws\credentials` both contain a default profile?
if they do i would suggest renaming the sso profile in the config or creating a different profile locally with a different name than `default`.
Could you try this and let us know if this mitigates the issue.

**Comment by tomphan** [comment]
### Error no. 1 (Solved): 

> An error occurred (AccessDenied) when calling the ListBuckets operation

Since I do not use `S3 `as checking parameter as suggestion, just replaced the testing command by the `aws amplify list-apps` command for my case, the output displayed well a meaning result, indicates that my configure was good for an aws sso.

### Error no. 2 (Pending, need to solve):

I stucked the whole day until now doing debugging & checking any available solution on the Internet, but still no luck.

Command:
`npx ampx sandbox --profile default --debug`

> SSMCredentialsError: UnrecognizedClientException: The security token included in the request is invalid.
> Resolution: Make sure your AWS credentials are set up correctly and have permissions to call SSM:GetParameter
> Cause: The security token included in the request is invalid.
> [DEBUG] 2024-12-09T14:56:10.810Z: SSMCredentialsError: UnrecognizedClientException: The security token included in the request is invalid.
>     at FileWatchingSandbox.isBootstrapped (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:264:23)
>     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async FileWatchingSandbox.start (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:87:30)
>     at async Object.handler (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/backend-cli/lib/commands/sandbox/sandbox_command.js:80:9)
> [DEBUG] 2024-12-09T14:56:10.813Z: UnrecognizedClientException: The security token included in the request is invalid.
>     at throwDefaultError (d:\01_Dev\Civil\amplify-vite-react-test\node_modules\@smithy\smithy-client\dist-cjs\index.js:836:20)
>     at d:\01_Dev\Civil\amplify-vite-react-test\node_modules\@smithy\smithy-client\dist-cjs\index.js:845:5
>     at de_CommandError (d:\01_Dev\Civil\amplify-vite-react-test\node_modules\@aws-sdk\client-ssm\dist-cjs\index.js:7021:14)
>     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async d:\01_Dev\Civil\amplify-vite-react-test\node_modules\@smithy\middleware-serde\dist-cjs\index.js:35:20
>     at async d:\01_Dev\Civil\amplify-vite-react-test\node_modules\@smithy\core\dist-cjs\index.js:168:18
>     at async d:\01_Dev\Civil\amplify-vite-react-test\node_modules\@smithy\middleware-retry\dist-cjs\index.js:320:38
>     at async d:\01_Dev\Civil\amplify-vite-react-test\node_modules\@aws-sdk\middleware-logger\dist-cjs\index.js:34:22
>     at async FileWatchingSandbox.isBootstrapped (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:241:46)
>     at async FileWatchingSandbox.start (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:87:30)

My `C:\Users\USERNAME\.aws` at this time:

`C:\Users\USERNAME\.aws\config`:

> [default]
> sso_session = amplify-admin
> sso_account_id = <...>
> sso_role_name = amplify-policy
> region = us-west-1
> 
> [sso-session amplify-admin]
> sso_start_url = https://<...>.awsapps.com/start
> sso_region = us-west-1
> sso_registration_scopes = sso:account:access
> 
> [profile default]
> sso_session = amplify-admin
> sso_account_id = <...>
> sso_role_name = amplify-policy
> region = us-west-1
> 
> [civil2]
> sso_session = amplify-admin
> sso_account_id = <...>
> sso_role_name = amplify-policy
> region = us-west-1

Where [profile default], [civil2] are 2 sections manually added by me for testing other profiles. <...> are sensitive info blurred when posting here.

`C:\Users\USERNAME\.aws\credentials`:

> [default]
> aws_access_key_id = <...>
> aws_secret_access_key = <...>
> aws_session_token = <...>

So, I tried created other profile than default, named "civil" and "civil2", but still no luck. Steps for reproductions:

Create a new profile named "civil3":

`aws configure sso`

> SSO session name [amplify-admin]:
> Attempting to automatically open the SSO authorization page in your default browser.
> If the browser does not open or you wish to use a different device to authorize this request, open the following URL:
> 
> https://oidc.us-west-1.amazonaws.com/authorize?response_type=code&client_id=zK88J<...>
> The only AWS account available to you is: <...>
> Using the account ID <...>
> The only role available to you is: amplify-policy
> Using the role name "amplify-policy"
> CLI default client Region [us-west-1]:
> CLI default output format [None]:
> CLI profile name [amplify-policy-<...>]: civil3
> 
> To use this profile, specify the profile name using --profile, as shown:
> 
> aws s3 ls --profile civil3

Check `C:\Users\USERNAME\.aws\config` again:

New section appeared:

> [profile civil3]
> sso_session = amplify-admin
> sso_account_id = <...>
> sso_role_name = amplify-policy
> region = us-west-1

Try: `aws amplify list-apps --profile civil3`:

Returns a meaningful output result

> {
>     "apps": [
>         {
>             "appId": "...",
>         },
>         {
>             "appId": "...",
>         }
>     ]
> }


`d:\<my dir>\amplify-vite-react-test>npx ampx sandbox --profile civil3 --debug`

> InvalidCredentialError: Failed to load AWS credentials for profile 'civil3'
> Resolution: To configure a new Amplify profile, use npx ampx configure profile.
> Cause: Could not load credentials from any providers
> [DEBUG] 2024-12-09T15:25:53.983Z: InvalidCredentialError: Failed to load AWS credentials for profile 'civil3'
>     at ensureAwsCredentialAndRegion (file:///d:/<my dir>/amplify-vite-react-test/node_modules/@aws-amplify/backend-cli/lib/command_middleware.js:44:19)
>     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async Promise.all (index 1)
> [DEBUG] 2024-12-09T15:25:53.985Z: CredentialsProviderError: Could not load credentials from any providers
>     at d:\<my dir>\amplify-vite-react-test\node_modules\@aws-sdk\credential-provider-node\dist-cjs\index.js:136:13
>     at d:\<my dir>\amplify-vite-react-test\node_modules\@smithy\property-provider\dist-cjs\index.js:97:33
>     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async coalesceProvider (d:\<my dir>\amplify-vite-react-test\node_modules\@smithy\property-provider\dist-cjs\index.js:124:18)
>     at async d:\01_Dev\Civil\amplify-vite-react-test\node_modules\@smithy\property-provider\dist-cjs\index.js:142:18
>     at async ensureAwsCredentialAndRegion (file:///<my dir>/amplify-vite-react-test/node_modules/@aws-amplify/backend-cli/lib/command_middleware.js:36:13)
>     at async Promise.all (index 1)

Please advise.

**Comment by tomphan** [comment]
Please note that the `C:\Users\USERNAME\.aws\credentials` file still having only one `default` section. No `civil3` section be shown.

> [default]
> aws_access_key_id = <...>
> aws_secret_access_key = <...>
> aws_session_token = <...>

**Comment by tomphan** [comment]
Got suggestion from the error output message that I should do the command: `npx ampx configure profile`, I haven't done that at this time, first trying your quick test suggestion:

```
d:\<my_dir>\amplify-vite-react-test>aws configure list --profile amplify-admin
      Name                    Value             Type    Location
      ----                    -----             ----    --------
   profile            amplify-admin           manual    --profile

The config profile (amplify-admin) could not be found

d:\<my_dir>\amplify-vite-react-test>aws configure list --profile default
      Name                    Value             Type    Location
      ----                    -----             ----    --------
   profile                  default           manual    --profile
access_key     ****************5IOO              sso
secret_key     ****************JqEL              sso
    region                us-west-1      config-file    ~/.aws/config

d:\<my_dir>\amplify-vite-react-test>aws configure list --profile civil3
      Name                    Value             Type    Location
      ----                    -----             ----    --------
   profile                   civil3           manual    --profile
access_key     ****************5IOO              sso
secret_key     ****************JqEL              sso
    region                us-west-1      config-file    ~/.aws/config

```

**Comment by tomphan** [comment]
Trying additional:

1. Renamed the whole `.aws` folder to another backup one.
2. Call `d:\<my_dir>\amplify-vite-react-test>aws configure sso`:

> d:\<my_dir>\amplify-vite-react-test>aws configure sso
> 
> SSO session name (Recommended): amplify-admin
> SSO start URL [None]: https://<...>.awsapps.com/start/#
> SSO region [None]: us-west-1
> SSO registration scopes [sso:account:access]:
> Attempting to automatically open the SSO authorization page in your default browser.
> If the browser does not open or you wish to use a different device to authorize this request, open the following URL:
> 
> http<...>
> The only AWS account available to you is: <...>
> Using the account ID <...>
> The only role available to you is: amplify-policy
> Using the role name "amplify-policy"
> CLI default client Region [None]: us-west-1
> CLI default output format [None]:
> CLI profile name [amplify-policy-<...>]: civil5
> 
> To use this profile, specify the profile name using --profile, as shown:
> 
> aws s3 ls --profile civil5

Testing: 
`d:\<my_dir>\amplify-vite-react-test>aws amplify list-apps --profile civil5`

Output meaningful JSON result.

Content of `c:\Users\<MY USERNAME>\.aws\`:
```
09/12/2024  10:59 PM    <DIR>          cli
09/12/2024  10:55 PM    <DIR>          sso
09/12/2024  10:56 PM               284 config
```

Note that only `config `file generated. No `credentials `file created.

Content of `config `file:

> [profile civil5]
> sso_session = amplify-admin
> sso_account_id = <...>
> sso_role_name = amplify-policy
> region = us-west-1
> [sso-session amplify-admin]
> sso_start_url = https://<...>.awsapps.com/start/#
> sso_region = us-west-1
> sso_registration_scopes = sso:account:access
> 

Try the sandbox command:

`d:\<my_dir>\amplify-vite-react-test>npx ampx sandbox --profile civil5 --debug`

> ampx sandbox
> 
> Starts sandbox, watch mode for Amplify backend deployments
> 
> Commands:
> ...
> 
> Options:
> ...
> 
> InvalidCredentialError: Failed to load AWS credentials for profile 'civil5'
> Resolution: To configure a new Amplify profile, use npx ampx configure profile.
> Cause: Could not load credentials from any providers
> [DEBUG] 2024-12-09T16:05:06.606Z: InvalidCredentialError: Failed to load AWS credentials for profile 'civil5'
>     at ensureAwsCredentialAndRegion (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/backend-cli/lib/command_middleware.js:44:19)
>     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async Promise.all (index 1)
> [DEBUG] 2024-12-09T16:05:06.615Z: CredentialsProviderError: Could not load credentials from any providers
>     at d:\<my_dir>\amplify-vite-react-test\node_modules\@aws-sdk\credential-provider-node\dist-cjs\index.js:136:13
>     at d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\property-provider\dist-cjs\index.js:97:33
>     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async coalesceProvider (d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\property-provider\dist-cjs\index.js:124:18)
>     at async d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\property-provider\dist-cjs\index.js:142:18
>     at async ensureAwsCredentialAndRegion (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/backend-cli/lib/command_middleware.js:36:13)
>     at async Promise.all (index 1)
>     
    
Error appeared, saying I should do `npx ampx configure profile`. Doing so:

`d:\<my_dir>\amplify-vite-react-test>npx ampx configure profile --name civil5
`

> d:\<my_dir>\amplify-vite-react-test>npx ampx configure profile --name civil5
> 
> ? Do you already have IAM User credentials? y
> ? Enter Access Key ID:
> ? Enter Secret Access Key:
> ? Enter the AWS region to use with the 'civil5' profile (eg us-east-1, us-west-2, etc): us-west-1
> Created profile civil5 successfully!
> 

Another Error message about Security Token invalid (as mentioned in the previous comment)

`d:\<my_dir>\amplify-vite-react-test>npx ampx sandbox --profile civil5 --debug`

> ...
> SSMCredentialsError: UnrecognizedClientException: The security token included in the request is invalid.
> Resolution: Make sure your AWS credentials are set up correctly and have permissions to call SSM:GetParameter
> Cause: The security token included in the request is invalid.
> [DEBUG] 2024-12-09T16:09:41.606Z: SSMCredentialsError: UnrecognizedClientException: The security token included in the request is invalid.
>     at FileWatchingSandbox.isBootstrapped (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:264:23)
>     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async FileWatchingSandbox.start (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:87:30)
>     at async Object.handler (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/backend-cli/lib/commands/sandbox/sandbox_command.js:80:9)
> [DEBUG] 2024-12-09T16:09:41.609Z: UnrecognizedClientException: The security token included in the request is invalid.
>     at throwDefaultError (d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\smithy-client\dist-cjs\index.js:836:20)
>     at d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\smithy-client\dist-cjs\index.js:845:5
>     at de_CommandError (d:\<my_dir>\amplify-vite-react-test\node_modules\@aws-sdk\client-ssm\dist-cjs\index.js:7021:14)
>     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\middleware-serde\dist-cjs\index.js:35:20
>     at async d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\core\dist-cjs\index.js:168:18
>     at async d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\middleware-retry\dist-cjs\index.js:320:38
>     at async d:\<my_dir>\amplify-vite-react-test\node_modules\@aws-sdk\middleware-logger\dist-cjs\index.js:34:22
>     at async FileWatchingSandbox.isBootstrapped (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:241:46)
>     at async FileWatchingSandbox.start (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:87:30)
> 

So, there were 2 error messages `"Failed to load AWS credentials for profile 'civil5'"` and `"The security token included in the request is invalid"` based on my attempted to configure. 

Please still noted that, in the `c:\Users\<MY USERNAME>\.aws\` at this time, there is only `config` file existed, the `credentials` file still not created following the reproduction steps above.

Please advise.

**Comment by tomphan** [comment]
`d:\<my-dir>\amplify-vite-react-test>aws configure list --profile civil5`

```
      Name                    Value             Type    Location
      ----                    -----             ----    --------
   profile                   civil5           manual    --profile
access_key     ****************7X4R              sso
secret_key     ****************rZtl              sso
    region                us-west-1      config-file    ~/.aws/config

```

**Comment by tomphan** [comment]
> @tomphan could you quickly run `aws sso login --profile civil5`, then retry sandbox? wondering if the sso was not logged in causing this error

`d:\<my_dir>\amplify-vite-react-test>aws sso login --profile civil5`

> Attempting to automatically open the SSO authorization page in your default browser.
> If the browser does not open or you wish to use a different device to authorize this request, open the following URL:
> 
> https...........
> Successfully logged into Start URL: https://<...>.awsapps.com/start/#
> 

Try sandbox:
`d:\<my_dir>\amplify-vite-react-test>npx ampx sandbox --profile civil5 --debug`

> ...
> 
> SSMCredentialsError: UnrecognizedClientException: The security token included in the request is invalid.
> Resolution: Make sure your AWS credentials are set up correctly and have permissions to call SSM:GetParameter
> Cause: The security token included in the request is invalid.
> [DEBUG] 2024-12-09T16:28:54.917Z: SSMCredentialsError: UnrecognizedClientException: The security token included in the request is invalid.
>     at FileWatchingSandbox.isBootstrapped (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:264:23)
>     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async FileWatchingSandbox.start (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:87:30)
>     at async Object.handler (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/backend-cli/lib/commands/sandbox/sandbox_command.js:80:9)
> [DEBUG] 2024-12-09T16:28:54.921Z: UnrecognizedClientException: The security token included in the request is invalid.
>     at throwDefaultError (d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\smithy-client\dist-cjs\index.js:836:20)
>     at d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\smithy-client\dist-cjs\index.js:845:5
>     at de_CommandError (d:\<my_dir>\amplify-vite-react-test\node_modules\@aws-sdk\client-ssm\dist-cjs\index.js:7021:14)
>     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\middleware-serde\dist-cjs\index.js:35:20
>     at async d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\core\dist-cjs\index.js:168:18
>     at async d:\<my_dir>\amplify-vite-react-test\node_modules\@smithy\middleware-retry\dist-cjs\index.js:320:38
>     at async d:\<my_dir>\amplify-vite-react-test\node_modules\@aws-sdk\middleware-logger\dist-cjs\index.js:34:22
>     at async FileWatchingSandbox.isBootstrapped (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:241:46)
>     at async FileWatchingSandbox.start (file:///d:/<my_dir>/amplify-vite-react-test/node_modules/@aws-amplify/sandbox/lib/file_watching_sandbox.js:87:30)
> 


I guess the credentials sent from sandbox was not come from the sso, but came from my `npx ampx configure profile --name civil5` settings. The error message appeared only after I set `npx ampx configure profile --name civil5` and entered some `Access Key ID` and `Secret Access Key` manually. I be noticed that these Access Key ID & Secret were different from the SSO listing, I got those Keys from the `AWS access portal` website, `AWS accounts` -> `Access keys`, not from any SSO info.

**Comment by tomphan** [comment]
So for clarify, for SSO login credentials, the error message was:

> InvalidCredentialError: Failed to load AWS credentials for profile 'civil6'
> Resolution: To configure a new Amplify profile, use npx ampx configure profile.
> Cause: Could not load credentials from any providers

For `npx ampx configure profile --name civil5` credentials (I don't know if I pasted the credentials correctly because the command line credentials was masked as not shown in the prompt, and I cannot find any Environement variable or text config file to double check the input), the error message was:

> SSMCredentialsError: UnrecognizedClientException: The security token included in the request is invalid.
> Resolution: Make sure your AWS credentials are set up correctly and have permissions to call SSM:GetParameter
> Cause: The security token included in the request is invalid.

**Comment by ykethan** [comment]
@tomphan after the new local sso profile was setup was `aws sso login --profile ` command executed?
I would also try opening a terminal session, rerunning the `aws sso login --profile ` command` and then running sandbox?

**Comment by tomphan** [comment]
@ykethan No luck, I can confirm the SSO credentials cannot passed into the providers chain, due to code programming bug in the SDK.

By installing the 3rd party logger:

`d:\<my_dir>\amplify-vite-react-test\node_modules\@aws-amplify\backend-cli\lib\`:

```
import log4js from "log4js";

log4js.configure({
  appenders: { out: { type: "stdout" } },
  categories: { default: { appenders: ["out"], level: "debug" } },
});

const logger = log4js.getLogger();
```

Pass the logger into the `fromNodeProviderChain` params:

```
        try {          
            await fromNodeProviderChain({
                ignoreCache: true,
                logger: logger
            })();
        }
```

I can ensure the SSO intput cannot passed into the code.
Debug log output:

```
d:\<my_dir>\amplify-vite-react-test>npx ampx sandbox --profile civil9
[2024-12-10T00:50:05.886] [DEBUG] default - @smithy/property-provider -> AWS_PROFILE is set, skipping fromEnv provider.
[2024-12-10T00:50:05.889] [DEBUG] default - @aws-sdk/credential-provider-node - defaultProvider::fromSSO
[2024-12-10T00:50:05.893] [DEBUG] default - @smithy/property-provider -> Skipping SSO provider in default chain (inputs do not include SSO fields).
[2024-12-10T00:50:05.900] [DEBUG] default - @aws-sdk/credential-provider-node - defaultProvider::fromIni
[2024-12-10T00:50:05.903] [DEBUG] default - @aws-sdk/credential-provider-ini - fromIni
[2024-12-10T00:50:05.911] [DEBUG] default - @smithy/property-provider -> Could not resolve credentials using profile: [civil9] in configuration/credentials file(s).
[2024-12-10T00:50:05.913] [DEBUG] default - @aws-sdk/credential-provider-node - defaultProvider::fromProcess
[2024-12-10T00:50:05.916] [DEBUG] default - @aws-sdk/credential-provider-process - fromProcess
[2024-12-10T00:50:05.918] [DEBUG] default - @smithy/property-provider -> Profile civil9 could not be found in shared credentials file.
[2024-12-10T00:50:05.919] [DEBUG] default - @aws-sdk/credential-provider-node - defaultProvider::fromTokenFile
[2024-12-10T00:50:05.926] [DEBUG] default - @aws-sdk/credential-provider-web-identity - fromTokenFile
[2024-12-10T00:50:05.929] [DEBUG] default - @smithy/property-provider -> Web identity configuration not specified
[2024-12-10T00:50:05.930] [DEBUG] default - @aws-sdk/credential-provider-node - defaultProvider::remoteProvider
[2024-12-10T00:50:05.933] [DEBUG] default - @aws-sdk/credential-provider-node - remoteProvider::fromInstanceMetadata
[2024-12-10T00:50:06.954] [DEBUG] default - AWS SDK Instance Metadata using v1 fallback (initial)
[2024-12-10T00:50:06.955] [DEBUG] default - @smithy/property-provider -> AWS_EC2_METADATA_V1_DISABLED not set in env, checking config file next.
[2024-12-10T00:50:07.961] [DEBUG] default - @smithy/property-provider (!) Could not load credentials from any providers
```

`d:\<my_dir>\amplify-vite-react-test\node_modules\@aws-sdk\credential-provider-node\dist-cjs\`:

in the code, I set a debug output of the init var:

```
      const { ssoStartUrl, ssoAccountId, ssoRegion, ssoRoleName, ssoSession } = init;
      _a.debug(init);
```

The `init` variable does not contain any SSO input. So I'm be sure no SSO credentials passed into the SDK code.

Do you think this is a programming bug from the SDK?

**Comment by ykethan** [comment]
@tomphan thank you for the information.
noticed `[2024-12-10T00:50:05.886] [DEBUG] default - @smithy/property-provider -> AWS_PROFILE is set, skipping fromEnv provider.`

by any chance is `AWS_PROFILE` env variable set on the terminal. Could you echo this variable and check if this is a different profile that may be getting picked up by the sandbox run.
if could try unsetting this env variable then try running the sandbox

**Comment by ykethan** [comment]
Hey @tomphan, would you be open for a quick call on this? i am currently available on discord my handle is `ykethan`

**Comment by ykethan** [comment]
thank you for hopping on call on this and working with me on this. The issue appear to be due to `credential-provider` on the sdk when using cmd. I would suggest filing an issue on the AWS SDK repository for further investigation.
https://github.com/aws/aws-sdk-js-v3

Closing the issue, do reach out on a new issue if you require any assistance

**Comment by nermin99** [comment]
@tomphan @ykethan What about for someone who is on a mac? I am getting the same issue

**Comment by nermin99** [comment]
Nvm I got it working by pasting the credentials directly into the shell:

`eval "$(aws configure export-credentials --profile default --format env)"
npx ampx sandbox`
